### PR TITLE
Scatter Plot Chart Gradient

### PR DESCRIPTION
**Pull request by jbrennan**
This PR extends the Scatter Plot rendering system to support gradients.

**Please remember this repository and PR discussion are public**

### CODE REVIEW

File: Package.swift [hunk 1]
```diff
@@ -1,13 +1,13 @@
...
 // The swift-tools-version declares the minimum version of Swift required to build this package.
 import PackageDescription
 
 let package = Package(
     name: "Charts",
     platforms: [
-          .iOS(.v9),
-          .tvOS(.v9),
-          .macOS(.v10_11),
+        .iOS("13.0"),
```
Comment (gemmakbarlow) [inline Package.swift]:
```suggestion
        .iOS("14.0"),
```
Optional: we could consider 14.0 if desired, due to bumping our minimum version recently (same for tvOS).

File: Source/Charts/Renderers/Scatter/ShapeRenderer.swift [hunk 2]
```diff
@@ -28,5 +28,17 @@ public protocol ShapeRenderer: AnyObject
         dataSet: ScatterChartDataSetProtocol,
         viewPortHandler: ViewPortHandler,
         point: CGPoint,
-        color: NSUIColor)
+        color: NSUIColor
+    )
+}
+
+/// Peloton-added Protocol, implemented by the CircleShaperRenderer to draw gradient circles.
```
Comment (gemmakbarlow) [inline Source/Charts/Renderers/Scatter/ShapeRenderer.swift]:
```suggestion
/// Peloton-added Protocol, implemented by the CircleShapeRenderer to draw gradient circles.
```

File: Source/Charts/Renderers/Scatter/CircleShapeRenderer.swift [hunk 3]
```diff
@@ -60,4 +61,68 @@ open class CircleShapeRenderer : NSObject, ShapeRenderer
             context.fillEllipse(in: rect)
         }
     }
+    
+    func renderGradientShape(
```
Comment (gemmakbarlow) [inline Source/Charts/Renderers/Scatter/CircleShapeRenderer.swift:65]:
```suggestion
    func renderGradientShape(
```
```suggestion
    open func renderGradientShape(
```

Optional: Suggest we match the visibility of the method above.

Comment (gemmakbarlow) [inline Source/Charts/Renderers/Scatter/CircleShapeRenderer.swift:65]:
```suggestion
    func renderGradientShape(
```
```suggestion
    open func renderGradientShape(
```

Optional: Suggest we match the visibility of the method above.

**Review by gemmakbarlow — APPROVED** [review]
✅

File: Source/Charts/Renderers/Scatter/CircleShapeRenderer.swift (hunk 3 above)
Comment (jbrennan) [reply Source/Charts/Renderers/Scatter/CircleShapeRenderer.swift:65]: Ahh I actually left this as internal intentionally, it's not API I wanted to expose outside of the module

**Review by TheNorthEestern — APPROVED** [review]

**Pull request merged**